docker/device_startup.docker

```diff
@@ -1,54 +1,50 @@
 #==============================================================================
-#Install everything required to run the camera server and run it
+#Install everything required to run a Huntsman device.
 #==============================================================================
 #Specify the OS
 FROM raspbian/jessie
 
-#Environment variables
+#Add the huntsman user
+ENV PANUSER=huntsman
+ENV PANDIR=/var/huntsman
+RUN useradd $PANUSER && mkdir $PANDIR && chown $PANUSER $PANDIR
+
+#Extra environment variables
 ENV DEBIAN_FRONTEND=noninteractive
 
 #Update apt-get 
-RUN apt-get -y upgrade
-RUN apt-get update
-RUN apt-get -y dist-upgrade
+RUN apt-get -y upgrade && apt-get update && apt-get -y dist-upgrade
 
 #Some initial packages
-RUN apt-get -y install bzip2
-RUN apt-get -y install git
-RUN apt-get -y install gcc
-RUN apt-get -y install sshfs 
+RUN apt-get -y install bzip2 git gcc sshfs 
 
 #==============================================================================
-#Berryconda
+#Berryconda (install as $PANUSER)
 
+USER $PANUSER
+WORKDIR $PANDIR
 ENV CONDA_URL=https://github.com/jjhelmus/berryconda/releases/download/v2.0.0/Berryconda3-2.0.0-Linux-armv7l.sh
-ENV CONDA_PATH=/root/berryconda3
-RUN wget $CONDA_URL
-RUN chmod +x Berryconda3-2.0.0-Linux-armv7l.sh
-RUN ./Berryconda3-2.0.0-Linux-armv7l.sh -b -p $CONDA_PATH
+ENV CONDA_PATH=/$PANDIR/berryconda3
+RUN wget $CONDA_URL && chmod +x Berryconda3-2.0.0-Linux-armv7l.sh && \
+    ./Berryconda3-2.0.0-Linux-armv7l.sh -b -p $CONDA_PATH
 
-RUN ln -s $CONDA_PATH/bin/conda /usr/local/bin/
-RUN ln -s $CONDA_PATH/bin/python3.6 /usr/local/bin/python
-RUN ln -s $CONDA_PATH/bin/pip /usr/local/bin/
+USER root
+RUN ln -s $CONDA_PATH/bin/conda /usr/local/bin/ && \
+    ln -s $CONDA_PATH/bin/python3.6 /usr/local/bin/python && \
+    ln -s $CONDA_PATH/bin/pip /usr/local/bin/
 
 RUN pip install --upgrade pip
 
 #==============================================================================
 #Huntsman-POCS
 
-RUN useradd huntsman
-
 #Huntsman-POCS variables
-ENV PANUSER=huntsman
-ENV PANDIR=/var/huntsman
 ENV POCS=${PANDIR}/POCS
 ENV PAWS=${PANDIR}/PAWS
 ENV HUNTSMAN_POCS=${PANDIR}/huntsman-pocs
 
 #POCs folders
-RUN mkdir $PANDIR && chown $PANUSER $PANDIR
-RUN mkdir -p $PANDIR/logs
-RUN mkdir -p $PANDIR/images
+RUN mkdir -p $PANDIR/logs && mkdir -p $PANDIR/images
```

Comment:
what happens to the logs in `$PANDIR/logs` after this docker instance is shutdown? Are the logs lost forever? Maybe we should make them a `sshfs` access point like you do for the `$PANDIR/images` directory?

Reply:
The way that I have been doing things is to map the host `$PANDIR` to the `$PANDIR` in the container. This is actually how you use it as a development mode as well. So really all of `$PANDIR` comes from the host and all of the system dependencies exist in the docker image.

Reply:
This makes sense. It also might solve a question for how to link up the to-be-developed-soon control computer docker instance to the yaml files created on-the-fly from @Physarah 's EMS: https://github.com/AstroHuntsman/huntsman-ems

Reply:
@lspitler I imagined those files to be distributed using the config server?